### PR TITLE
Dev workflow documentation updates

### PR DESCRIPTION
**Pull request by salonee13**
### What changes were proposed in this pull request?
Added lines to documentation to have the contributor check to see if they have yarn and node installed correctly. 

### How was this pull request tested?
Reviewed the page after make docs 

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Review by kiersten-stokes — APPROVED** [review]
LGTM! This will be good to have back in the docs to streamline future installations. Thanks!

**Review by kevin-bates — COMMENTED** [review]
This PR looks great.  I'm wondering, though, if we could add one more thing that just came up in [this discussion](https://github.com/elyra-ai/elyra/discussions/2852)?  If not, no big deal - it could be tackled another time.

I think this would amount to:
- adding a link to [`GNU Make`](https://www.gnu.org/software/make/) to the Requirements section.
- adding installation instructions - which would probably be best to just defer to the previously linked web page for their platform.
- adding a verification step to run `make` from the github directory (and reference the output that is a couple sections below in [Build & Installation](https://elyra.readthedocs.io/en/latest/developer_guide/development-workflow.html#build-installation)

**Comment by akchinSTC** [comment]
Sounds fair, `Make` should be included as a required utility, maybe include a small non-support disclaimer for windows and only include link for installation in *nix environments

**Comment by akchinSTC** [comment]
@salonee13 - you're probably busy making changes to the community repository. Let us know if you are unable to get to this before semester end and we can merge as is.

File: docs/source/developer_guide/development-workflow.md
Comment (kevin-bates) [inline docs/source/developer_guide/development-workflow.md:86]: I think this may be confusing if they have yet to set up the repository, which is the next step.  If there is no `Makefile` present, they will receive the message: `make: *** No targets specified and no makefile found.  Stop.`

Perhaps indicating both scenarios might be better, also making sure that `make` is invoked in the repository directory as well.

```suggestion
    To verify the installation, run `make`. 
    If you have yet to [set up the repository](#setting-up-your-development-environment), you should see a message like the following:
    ```
    make: *** No targets specified and no makefile found.  Stop.
    ```
    Once the repository is set up, running `make` from that location should display the available tasks that are listed in the [Build & Installation](#build-installation) section below.
```

**Review by kevin-bates — COMMENTED** [review]
This looks good.  Thank you for adding the information regarding `make`.  I had a small comment regarding the time and location in which the `make` installation is verified.

**Review by akchinSTC — APPROVED** [review]

**Pull request merged**
